### PR TITLE
E2e updates

### PR DESCRIPTION
**Pull request by yuval-k**
1. Use debug containers
2. Increase eventually polling time as the default 10ms is only suitable for channels
3. Use gibkgowriter everywhere to reduce verbosity
4. Add race detection

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,24 +46,27 @@ $(eval OUTPUT_BINARY := $(OUTPUT)/$(BINARY))
...
 $(OUTPUT_BINARY): $(OUTPUT) $(PREREQUISITES)
-	CGO_ENABLED=0 GOOS=linux go build -i -v -o $(OUTPUT_BINARY) cmd/$(BINARY)/main.go
+	CGO_ENABLED=0 GOOS=linux go build -v -o $(OUTPUT_BINARY) cmd/$(BINARY)/main.go
 $(OUTPUT_BINARY)-debug: $(OUTPUT) $(PREREQUISITES)
 	go build -i -gcflags "-N -l" -o $(OUTPUT_BINARY)-debug cmd/$(BINARY)/main.go
 
 # docker
 $(BINARY)-docker: $(OUTPUT_BINARY)
-	docker build -t $(DOCKER_USER)/$(BINARY):$(IMAGE_TAG) -f cmd/$(BINARY)/Dockerfile $(OUTPUT)
+	docker build -t $(DOCKER_USER)/$(BINARY):$(IMAGE_TAG) $(OUTPUT) -f - < cmd/$(BINARY)/Dockerfile
```
Comment (ilackarms) [inline Makefile]: what's the purpose of this chcange?

File: Makefile [hunk 2]
```diff
@@ -46,24 +46,27 @@ $(eval OUTPUT_BINARY := $(OUTPUT)/$(BINARY))
...
 .PHONY: $(BINARY)-docker-push
+.PHONY: $(BINARY)-docker-push-debug
 
 # nice targets for the binaries
 $(BINARY): $(OUTPUT_BINARY)
 $(BINARY)-debug: $(OUTPUT_BINARY)-debug
 
 # go build
 $(OUTPUT_BINARY): $(OUTPUT) $(PREREQUISITES)
-	CGO_ENABLED=0 GOOS=linux go build -i -v -o $(OUTPUT_BINARY) cmd/$(BINARY)/main.go
```
Comment (ilackarms) [inline Makefile]: why cant we have -i? because of the permissions thing?

File: Makefile (hunk 1 above)
Comment (yuval-k) [reply Makefile]: Docker refused to build if the docker file is not in the build context. This fixes it

File: cmd/control-plane/Dockerfile.debug [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.7
-COPY gloo-debug /gloo
+FROM ubuntu
+COPY control-plane-debug /control-plane-debug
```
Comment (ilackarms) [inline cmd/control-plane/Dockerfile.debug]: we probably want to rename `*-debug` to just `*` here (lose the debug) so the kubernetes yaml files etc. don't have to be changed

File: Makefile (hunk 2 above)
Comment (yuval-k) [reply Makefile]: Correct. I made sure the e2e tests can work with debug containers (via env vars) so you can that to iterate fast

File: cmd/control-plane/Dockerfile.debug (hunk 3 above)
Comment (yuval-k) [reply cmd/control-plane/Dockerfile.debug]: I changed them already to remove the command part (i.e use the entry point setting) it think it is better as it makes the kube yamls less fragile. Thoughts?

File: test/helpers/build_images.go [hunk 4]
```diff
@@ -35,7 +37,7 @@ func hash(h string) string {
 }
 
 // builds and pushes all docker containers needed for test
-func BuildPushContainers(push bool) error {
+func BuildPushContainers(push bool, debug bool) error {
```
Comment (ilackarms) [inline test/helpers/build_images.go]: nit: `push, debug bool`

File: Makefile (hunk 1 above)
Comment (ilackarms) [reply Makefile]: it works for me without this change. what's the difference?

File: Makefile (hunk 2 above)
Comment (ilackarms) [reply Makefile]: ok

File: cmd/control-plane/Dockerfile.debug (hunk 3 above)
Comment (ilackarms) [reply cmd/control-plane/Dockerfile.debug]: i'd rather still rename the binary

Comment (ilackarms) [reply cmd/control-plane/Dockerfile.debug]: makes it easier to transition between containers for testing, manual debug, etc.

File: Makefile (hunk 1 above)
Comment (yuval-k) [reply Makefile]: not sure.. maybe docker version? it fails on my machine

**Comment by ilackarms** [comment]
looks good!

**Pull request merged**